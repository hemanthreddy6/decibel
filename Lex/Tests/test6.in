import "fdsklj";

sum <- (a:int, b:int):int{
    return a + b;
}

add1 <- sum(_, 1);

main {
    a <- 0;
    a = add1(2);
    b <- sum(1);
    return a;
}
