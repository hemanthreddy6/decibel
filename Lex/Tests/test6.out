IMPORT
STRING_LITERAL
;
FUNCTION
IDENTIFIER
<-
(
IDENTIFIER
:
INT
,
IDENTIFIER
:
INT
)
:
INT
{
RETURN
IDENTIFIER
+
IDENTIFIER
;
}
FUNCTION
IDENTIFIER
<-
IDENTIFIER
(
_
,
INT_LITERAL
)
;
FUNCTION
IDENTIFIER
<-
(
)
{
IDENTIFIER
<-
INT_LITERAL
;
IDENTIFIER
=
IDENTIFIER
(
INT_LITERAL
)
;
IDENTIFIER
<-
IDENTIFIER
(
INT_LITERAL
)
;
RETURN
IDENTIFIER
;
}
